src/TrackGenerator.cpp

```diff
@@ -268,9 +273,27 @@ double TrackGenerator::getZCoord() {
 
 
 /**
+ * @brief Return the array used to store the FSR volumes
+ * @return _FSR_volumes the FSR volumes array indexed by FSR ID
+ */
+FP_PRECISION* TrackGenerator::getFSRVolumesBuffer() {
+#pragma omp critical
+  {
+    if (_FSR_volumes == NULL) {
+      int num_FSRs = _geometry->getNumFSRs();
+      _FSR_volumes = new FP_PRECISION[num_FSRs];
+      memset(_FSR_volumes, 0., num_FSRs*sizeof(FP_PRECISION));
+    }
+  }
+
+  return _FSR_volumes;
+}
+
+
+/**
  * @brief Computes and returns an array of volumes indexed by FSR.
- * @details Note: It is the function caller's responsibility to deallocate
- *          the memory reserved for the FSR volume array.
+ * @details Note: The memory is stored in the FSR volumes buffer of the
+ *          TrackGenerator and is freed during deconstruction.
```

Comment:
Maybe I should just take out the comment altogether. We don't mention how the memory is managed for other variables.